analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Pattern.cs

```diff
@@ -76,20 +76,24 @@ private static SymbolicConstraint ConstraintFromConstantPattern(ProgramState sta
                     return objectConstraint.ApplyOpposite(useOpposite);
                 }
             }
-            return null;
+            return isReferenceType ? ObjectConstraint.NotNull.ApplyOpposite(useOpposite) : null;    // "obj is 42" => "obj" is NotNull and obj.ToString() is safe. We don't have this for bool.
```

Comment:
> We don't have this for bool.

For a boxed bool, we have it:
`object b = true`;
b HasConstraints(BoolContraint.True, ObjectConstraint.NotNull)

Reply:
`object b = true` is tricky, but doable. And it's not related to declaration pattern.